### DISCUSSION TITLE
x-for  in range   no response

### DISCUSSION THREAD

**Discussion by lukexiaoyu** [alpinejs/alpine#3528] [1. Help] [answered]
```html
<ul>
    <template x-for="i in 10">
        <li x-text="i"></li>
    </template>
</ul>
```

the docs said x-for can write like this ,but  there's no response .and No error reported.
so why? is anyone know?

**Comment by joshhanley** [comment, marked as answer]
@lukexiaoyu you need `x-data` otherwise it won't work. Some of the examples are stripped down, so that's why it's not displayed.

Hope this helps!

**Reply by lukexiaoyu** [reply]
OK,thanks
